cmd/crank/beta/trace/trace.go

```diff
@@ -0,0 +1,164 @@
+)
+
+const (
+	errGetResource        = "cannot get requested resource"
+	errCliOutput          = "cannot print output"
+	errKubeConfig         = "failed to get kubeconfig"
+	errInitKubeClient     = "cannot init kubeclient"
+	errGetDiscoveryClient = "cannot get discovery client"
+	errGetMapping         = "cannot get mapping for resource"
+	errInitPrinter        = "cannot init new printer"
+)
+
+// Cmd builds the trace tree for a Crossplane resource.
+type Cmd struct {
+	Resource string `arg:"" help:"Kind of the of the Crossplane resource, accepts the 'TYPE[.VERSION][.GROUP]' format."`
+	Name     string `arg:"" help:"Name of the Crossplane resource."`
+
+	// TODO(phisco): add support for all the usual kubectl flags; configFlags := genericclioptions.NewConfigFlags(true).AddFlags(...)
+	// TODO(phisco): move to namespace defaulting to "" and use the current context's namespace
+	Namespace             string `short:"n" name:"namespace" help:"Namespace of the resource." default:"default"`
+	Output                string `short:"o" name:"output" help:"Output format. One of: default, wide, json, dot." enum:"default,wide,json,dot" default:"default"`
+	ShowConnectionSecrets bool   `short:"s" name:"show-connection-secrets" help:"Show connection secrets in the output."`
+}
+
+// Help returns help message for the trace command.
+func (c *Cmd) Help() string {
+	return `
+This command trace a Crossplane resource to get a detailed output of its
+relationships, helpful for troubleshooting.
+
+If needed the resource kind can be also specified further,
+'TYPE[.VERSION][.GROUP]', e.g. mykind.example.org.
+
+Examples:
+  # Trace a MyKind resource (mykinds.example.org/v1alpha1) named 'my-res' in the namespace 'my-ns'
+  crossplane beta trace mykind my-res -n my-ns
+
+  # Output wide format, showing full errors and condition messages
+  crossplane beta trace mykind my-res -n my-ns -o wide
+
+  # Show connection secrets in the output
+  crossplane beta trace mykind my-res -n my-ns --show-connection-secrets
+
+  # Output a graph in dot format and pipe to dot to generate a png
+  crossplane beta trace mykind my-res -n my-ns -o dot | dot -Tpng -o output.png
+
+  # Output all retrieved resources to json and pipe to jq to have it coloured
+  crossplane beta trace mykind my-res -n my-ns -o json | jq
+
+  # Output debug logs to stderr while redirecting a dot formatted graph to dot
+  crossplane beta trace mykind my-res -n my-ns -o dot --verbose | dot -Tpng -o output.png
+`
+}
+
+// Run runs the trace command.
+func (c *Cmd) Run(k *kong.Context, logger logging.Logger) error { //nolint:gocyclo // TODO(phisco): refactor
+	logger = logger.WithValues("Resource", c.Resource, "Name", c.Name)
+
+	// Init new printer
+	p, err := printer.New(c.Output)
+	if err != nil {
+		return errors.Wrap(err, errInitPrinter)
+	}
+	logger.Debug("Built printer", "output", c.Output)
+
+	kubeconfig, err := ctrl.GetConfig()
+	if err != nil {
+		logger.Debug(errKubeConfig, "error", err)
```

Comment:
nit: Do we need to log given we're already returning. And also to be consistent with other error returns.

Reply:
done: https://github.com/crossplane/crossplane/pull/4849/commits/f8998ceee913e53c1c4585e448f5d0936c44fb38